hbase-server/src/test/java/org/apache/hadoop/hbase/client/TestSeparateClientZKCluster.java

```diff
@@ -255,7 +260,20 @@ public void testAsyncTable() throws Exception {
       Get get = new Get(row);
       Result result = table.get(get).get();
       LOG.debug("Result: " + Bytes.toString(result.getValue(family, qualifier)));
-      Assert.assertArrayEquals(value, result.getValue(family, qualifier));
+      assertArrayEquals(value, result.getValue(family, qualifier));
+    }
+  }
+
+  @Test
+  public void testChangeMetaReplicaCount() throws Exception {
```

Comment:
Maybe adding more logics to verify the `ClientZKSyncer` is working well when meta replica changes, especially when reducing from multiple to single replica?

Reply:
In the test we increase from 1 to 3, and then reduce back to 2, you mean we should add one more assert to reduce to 1?

Reply:
Yes, and it seems the test now didn't test any `ClientZKSyncer` logics, like if the client could still correctly access meta through client ZK, but checks against meta region locator?

Reply:
RegionLocator will go to ConnectionRegistry to ask for the meta locations and we have already set the ConnectionRegistry to ZKConnectionRegistry against the client zk in the setup method, so it is testing the ClientZKSyncer logics, you could try comment out the set data logic in ClientZKSyncer, the test will fail with timeout...

Reply:
ok, got it, makes sense. Thanks for the clarification.